src/bootstrap.php

```diff
@@ -0,0 +1,293 @@
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Throwable;
+
+$config = array_merge($config ?? [], include __DIR__.'/../config.php');
+
+$app = new Application([
+    'api.uri' => $config['api.uri'] ?? 'https://api.elifesciences.org/',
+    'api.timeout' => $config['api.timeout'] ?? 1,
+    'debug' => $config['debug'] ?? false,
+    'logger.path' => $config['logger.path'] ?? __DIR__.'/../var/logs',
+    'logger.level' => $config['logger.level'] ?? LogLevel::WARNING,
+]);
+
+$app['elife.guzzle_client'] = function () use ($app) {
+    return new Client([
+        'base_uri' => $app['api.uri'],
+        'connect_timeout' => 0.5,
+        'timeout' => $app['api.timeout'],
+    ]);
+};
+
+$app['elife.api_client'] = function () use ($app) {
+    return new Guzzle6HttpClient($app['elife.guzzle_client']);
+};
+
+if ($app['debug']) {
+    $app->extend('elife.api_client', function (HttpClient $httpClient) use ($app) {
+        return new WarningCheckingHttpClient($httpClient, $app['logger']);
+    });
+}
+
+$app['elife.api_sdk'] = function () use ($app) {
+    return new ApiSdk($app['elife.api_client']);
+};
+
+$app['elife.api_sdk.serializer'] = function () use ($app) {
+    return $app['elife.api_sdk']->getSerializer();
+};
+
+$app['elife.logger.factory'] = function (Application $app) {
+    return new LoggingFactory($app['logger.path'], 'recommendations-api', $app['logger.level']);
+};
+
+$app['logger'] = function (Application $app) {
+    return $app['elife.logger.factory']->logger();
+};
+
+$app['negotiator'] = function () {
+    return new VersionedNegotiator();
+};
+
+$app->get('/recommendations/{type}/{id}', function (Request $request, string $type, string $id) use ($app) {
+    try {
+        $identifier = Identifier::fromString("{$type}/{$id}");
+
+        if ('article' !== $type) {
+            throw new BadRequestHttpException('Not an article');
+        }
+    } catch (InvalidArgumentException $e) {
+        throw new NotFoundHttpException();
+    }
+
+    $accepts = [
+        'application/vnd.elife.recommendations+json; version=1',
+    ];
+
+    /** @var Accept $type */
+    $type = $app['negotiator']->getBest($request->headers->get('Accept'), $accepts);
+
+    $version = (int) $type->getParameter('version');
+    $type = $type->getType();
+
+    $page = $request->query->get('page', 1);
+    $perPage = $request->query->get('per-page', 20);
+
+    $article = $app['elife.api_sdk']->articles()->getHistory($id);
+
+    // Do this early as we have to then request each episode.
+    $podcastEpisodes = $app['elife.api_sdk']->podcastEpisodes()
+        ->containing(Identifier::article($id))
+        ->slice(0, 100);
+
+    try {
+        $article = $article->wait()->getVersions()[0];
```

Comment:
Yep. I'm using the article versions endpoint as the article endpoint is a lot bigger size-wise, so I'm assuming it would be a little faster.